siobrultech_protocols/gem/api.py

```diff
@@ -38,13 +38,24 @@ def __init__(self, formatter: Callable[[T], str], parser: Callable[[str], R]):
         self._format_request = formatter
         self._parse_response = parser
 
-    def send_request(self, protocol: BidirectionalProtocol, arg: T) -> timedelta:
+    def send_request(
+        self,
+        protocol: BidirectionalProtocol,
+        arg: T,
+        serial_number: Optional[int] = None,
+    ) -> timedelta:
         """
         Send the request using the given protocol and argument.
 
         Returns the length of time the caller should wait before attempting to receive the response.
         """
-        return protocol.send_api_request(self._format_request(arg))
+        formatted_request = self._format_request(arg)
+        if serial_number:
+            formatted_request = formatted_request.replace(
+                "^^^", f"^^^NMB{serial_number%100000:05}"
```

Comment:
Can we add a comment to explain `NMB`?

Reply:
Added a constant with a descriptive name.